### ISSUE TITLE
BIT deletes source files

### ISSUE THREAD

**Issue by djpeter84** [bit-team/backintime#1069]
I  took snapshots on Dec 2019 and Jan 2020 and March 2020. 

Today, I was looking for some files on my Desktop (the Source) but they were missing. 

I looked at my old Snapshots and  found out that those missing files were also missing in the March 2020 backup but were in the Dec 2019 and jan 2020 backup. 

I ruled out that i accidentally and manually deleted those files because there were over 20 missing and there were in different locations. 

My best guess is that BIT deleted  some (not sure how many more) files from my Desktop (Source) that were backed up on Dec 2019 and jan 2020. 

So something happened between the Jan 2020 backup and the Backup I did in March.

Is there a setting in BIT that tells BIT to delete the file from source if it is backed up and unchanged in a previous snapshot?

**Comment by colinl** [comment]
Have a look in the BiT log for the first snapshot they are missing from and see if it mentions those files?

**Comment by djpeter84** [comment]
> Have a look in the BiT log for the first snapshot they are missing from and see if it mentions those files?

I looked at the log and yes those files (and many more were deleted). How do i prevent this from happening? is there something in the settings?

**Comment by colinl** [comment]
I presume from that that you saw lines like
` *deleting   home/user/path/to/file`
That means that BiT cannot find that file so is noting that it has been deleted from your system since the last backup.  What makes you think that it is BiT that deleted them rather than any other application that you run on your system?

**Comment by djpeter84** [comment]
Yes, i have lines like: [C] *deleting   mnt/BA3EDBB73EDB6AC2/ABN/03 PHASE 3  APPOINTMENT/OTHER2_INCLUDING EMAILS/☑ Does my life insurance beneficiary have to be a relative or can I select someone else?.html and [C] *deleting   mnt/BA3EDBB73EDB6AC2/ABN/03 PHASE 3  APPOINTMENT/Stats and References/How Much a Funeral Costs and Average Funeral Costs: Your Complete Guide.html 
I have not changed any app or seen any other unusual app running that would have deleted these files. It is possible but i don't think it is probable. The only other app that I can think of that scans folders/files is Recoll.

So i take it that there is no setting in BIT that deletes files from the source once they have been backed up?

**Comment by colinl** [comment]
`*deleting mnt/BA3EDBB73EDB6AC2/`
Does that imply it was on a removable drive of some sort?

**Comment by djpeter84** [comment]
Yes, i have lines like: [C] *deleting   mnt/BA3EDBB73EDB6AC2/ABN/03 PHASE 3  APPOINTMENT/OTHER2_INCLUDING EMAILS/☑ Does my life insurance beneficiary have to be a relative or can I select someone else?.html and [C] *deleting   mnt/BA3EDBB73EDB6AC2/ABN/03 PHASE 3  APPOINTMENT/Stats and References/How Much a Funeral Costs and Average Funeral Costs: Your Complete Guide.html 
I have not changed any app or seen any other unusual app running that would have deleted these files. It is possible but i don't think it is probable. The only other app that I can think of that scans folders/files is Recoll.

So i take it that there is no setting in BIT that deletes files from the source once they have been backed up?

> `*deleting mnt/BA3EDBB73EDB6AC2/`
> Does that imply it was on a removable drive of some sort?

My hard drive is partitioned. That's where I keep my personal files (separated from OS).

**Comment by colinl** [comment]
OK, not that then.  Is there any common factor between the missing files?

**Comment by djpeter84** [comment]
most seem to be

> OK, not that then. Is there any common factor between the missing files?

most seem to be html files.

**Comment by colinl** [comment]
Have you run a disc health check in case it is a hardware problem?
Also run fsck I suppose.

**Comment by djpeter84** [comment]
is there a way to just restore certain file types (e.g., html, jpeg) and not others?

**Comment by djpeter84** [comment]
UPDATE: i just ran BIT again and was able to reproduce the issue. BIT deleted files from HD that were present on HD.  I am including some examples along with the current BIT settings. The files were present in the last backup (3/27/2020) but were deleted. I also noticed that the backup took two days to run compared to the usual 8hrs or so and the log produced over 3,000 pages of changes when i copied the log into Word. I have not made that many changes to my desktop files so there should not be many files that changed since my last backup.

Again, my understanding is that BIT should not delete files from the source (e.g., desktop), unless one restores a backup....

I am also not sure why BIT is deleting some files (mostly html) and not others.

Example where file is present in last backup:

![file present in BIT](https://user-images.githubusercontent.com/39010720/81476347-d15c5080-91ce-11ea-8a45-0563746dc3fb.jpg)

File is now gone from current backup and hard drive/current system:

![file deleted from drive and snapshot](https://user-images.githubusercontent.com/39010720/81476363-e5a04d80-91ce-11ea-845d-b3bcac961249.jpg)

BIT log showing file was deleted:

![snapshot log example](https://user-images.githubusercontent.com/39010720/81476370-e9cc6b00-91ce-11ea-8650-85dd5bd71e64.jpg)

Another BIT log showing file was deleted:
![snapshot log example2](https://user-images.githubusercontent.com/39010720/81476376-ec2ec500-91ce-11ea-9ad7-9e5401d5b901.jpg)

My current BIT settings:

![BITsettings1](https://user-images.githubusercontent.com/39010720/81476379-ef29b580-91ce-11ea-9552-f3244d664726.jpg)

![BITsettings2](https://user-images.githubusercontent.com/39010720/81476383-f2bd3c80-91ce-11ea-9b9c-f72d1e747c0d.jpg)

![BITsettings3](https://user-images.githubusercontent.com/39010720/81476388-f650c380-91ce-11ea-9376-76a5cd9ada1a.jpg)

![BITsettings4](https://user-images.githubusercontent.com/39010720/81476389-f8b31d80-91ce-11ea-9e84-fbcf0a28ae0d.jpg)

![BITsettings5](https://user-images.githubusercontent.com/39010720/81476394-fb157780-91ce-11ea-8c83-ece06d53c310.jpg)

![BITsettings6](https://user-images.githubusercontent.com/39010720/81476395-fe106800-91ce-11ea-9a26-3f9655d09b74.jpg)

BIT total list of changes after i copied log into Word (also did a find for "deleting" in Word):

![BIT total list of changes](https://user-images.githubusercontent.com/39010720/81476398-0072c200-91cf-11ea-8c19-07a8b502cb76.jpg)

Another Example where file is present in last backup:

![file present in BIT2](https://user-images.githubusercontent.com/39010720/81476399-023c8580-91cf-11ea-9c07-d03fd09b943f.jpg)

File is now deleted:

![file deleted from drive and snapshot2](https://user-images.githubusercontent.com/39010720/81476406-049edf80-91cf-11ea-90c9-85184b9e5477.jpg)

BIT log showing file was deleted:

![snapshot log example3](https://user-images.githubusercontent.com/39010720/81476407-0668a300-91cf-11ea-90f5-feb07c13e498.jpg)

**Comment by emtiu** [comment]
This is still a worrying report, but we haven't heard from the reporter in a while. One thing that catches my eye is that the affected files all have special characters in their filenames, such as:
- `☑️`
- `?`
- `:`
- `|`

This might be confusing rsync (or some other tool involved, like apparently veracrypt). In any case, though, there shouldn't ever be a situation where _taking a snaphot_ causes _deletions at the source_.

@djpeter84 Can you still reproduce the problem?

**Comment by djpeter84** [comment]
Forgot about this post. After some troubleshooting, I suspect that the cause was Windows CHKDSK.
I had ran CHKDSK right before I ran BIT sometime after to backup my files and then I found that files were deleted around the same time. Since BIT was set to sync with my desktop (source), it also deleted those files from the backup.

 
https://www.minitool.com/data-recovery/recover-data-after-chkdsk.html
https://www.minitool.com/data-recovery/recover-chkdsk-deleted-files.html
